### DISCUSSION TITLE
Does anyone understand Discord's image embed URL format?

### DISCUSSION THREAD

**Discussion by Newbrict** [Rapptz/discord.py#9449] [Q&A] [answered]
I'm not sure if this belongs here, but I don't know where else to ask.

The url format is like so:
`https://cdn.discordapp.com/attachments/{channel_id}/{????????}/{filename.extension}`

Does anyone know what that second id is? The "???????" in my example. I cannot find it documented anywhere.

**Comment by LeoCx1000** [comment]
~~Message ID~~ oh

**Comment by LostLuma** [comment]
It's the attachment ID which is unique to every attachment, generated with a timestamp very similar to the message it belongs to.

**Reply by Newbrict** [reply]
Is there any way to recover that ID for a deleted file?

**Reply by LostLuma** [reply, marked as answer]
Not unless you've logged it anywhere, but it's also not useful once an attachment is deleted.

**Reply by Newbrict** [reply]
Thanks.
